tests/Unit/PointwiseFunctions/AnalyticData/CurvedWaveEquation/Test_ScalarWaveGr.cpp

```diff
@@ -0,0 +1,402 @@
+template <typename ScalarFieldData>
+void test_no_hole(
+    const CurvedScalarWave::AnalyticData::ScalarWaveGr<
+        ScalarFieldData, gr::Solutions::KerrSchild>& curved_wave_data,
+    const ScalarFieldData& flat_wave_solution) noexcept {
+  const tnsr::I<DataVector, 3> x{std::array<DataVector, 3>{
+      {DataVector({0., 1., 2., 3.}), DataVector({0., 0., 0., 0.}),
+       DataVector({0., 0., 0., 0.})}}};
+  auto vars = curved_wave_data.variables(
+      x, tmpl::list<CurvedScalarWave::Pi, CurvedScalarWave::Phi<3>,
+                    CurvedScalarWave::Psi>{});
+  auto flat_vars = flat_wave_solution.variables(
+      x, 0., tmpl::list<ScalarWave::Pi, ScalarWave::Phi<3>, ScalarWave::Psi>{});
+  CHECK_ITERABLE_APPROX(get(get<CurvedScalarWave::Psi>(vars)),
+                        get(get<ScalarWave::Psi>(flat_vars)));
+  CHECK_ITERABLE_APPROX(get(get<CurvedScalarWave::Pi>(vars)),
+                        get(get<ScalarWave::Pi>(flat_vars)));
+  CHECK_ITERABLE_APPROX(get<0>(get<CurvedScalarWave::Phi<3>>(vars)),
+                        get<0>(get<ScalarWave::Phi<3>>(flat_vars)));
+  CHECK_ITERABLE_APPROX(get<1>(get<CurvedScalarWave::Phi<3>>(vars)),
+                        get<1>(get<ScalarWave::Phi<3>>(flat_vars)));
+  CHECK_ITERABLE_APPROX(get<2>(get<CurvedScalarWave::Phi<3>>(vars)),
+                        get<2>(get<ScalarWave::Phi<3>>(flat_vars)));
+}
+
+template <typename ScalarFieldData>
+void test_kerr(
+    const CurvedScalarWave::AnalyticData::ScalarWaveGr<
+        ScalarFieldData, gr::Solutions::KerrSchild>& curved_wave_data,
+    const gr::Solutions::KerrSchild& bh_solution,
+    const ScalarFieldData& flat_wave_solution) noexcept {
+  const tnsr::I<DataVector, 3> x{std::array<DataVector, 3>{
+      {DataVector({0., 1., 2., 3.}), DataVector({0., 0., 0., 0.}),
+       DataVector({0., 0., 0., 0.})}}};
+  auto vars = curved_wave_data.variables(
+      x, tmpl::list<CurvedScalarWave::Pi, CurvedScalarWave::Phi<3>,
+                    CurvedScalarWave::Psi>{});
+
+  const auto flat_wave_vars = flat_wave_solution.variables(
+      x, 0., ScalarWave::System<3>::variables_tag::tags_list{});
+  const auto flat_wave_dt_vars = flat_wave_solution.variables(
+      x, 0.,
+      tmpl::list<::Tags::dt<ScalarWave::Pi>, ::Tags::dt<ScalarWave::Phi<3>>,
+                 ::Tags::dt<ScalarWave::Psi>>{});
+
+  const auto kerr_variables = bh_solution.variables(
+      x, 0., gr::Solutions::KerrSchild::tags<DataVector>{});
+
+  // construct the vars in-situ
+  const auto& local_psi = get<ScalarWave::Psi>(flat_wave_vars);
+  const auto& local_phi = get<ScalarWave::Phi<3>>(flat_wave_vars);
+  auto local_pi = make_with_value<Scalar<DataVector>>(x, 0.);
+  {
+    const auto shift_dot_dpsi = dot_product(
+        get<gr::Tags::Shift<3, Frame::Inertial, DataVector>>(kerr_variables),
+        get<ScalarWave::Phi<3>>(flat_wave_vars));
+    get(local_pi) = (get(shift_dot_dpsi) -
+                     get(get<::Tags::dt<ScalarWave::Psi>>(flat_wave_dt_vars))) /
```

Comment:
don't use the time derivative, it will get removed eventually

Reply:
updated, now using `Pi`.